dbms/src/Storages/DeltaMerge/SegmentReadTask.cpp

```diff
@@ -551,6 +551,7 @@ void SegmentReadTask::doFetchPages(const disaggregated::FetchDisaggPagesRequest
         cluster->rpc_client,
         extra_remote_info->store_address);
     grpc::ClientContext client_context;
+    rpc.setClientContext(client_context, dm_context->global_context.getSettingsRef().disagg_fetch_pages_timeout);
```

Comment:
If we set a timeout for a streaming call, the timeout is to limit the whole process time (`until stream_resp->Finish()` is called) rather than a timeout for each message.